packages-next/keystone/src/lib/core/field-assertions.ts

```diff
@@ -69,30 +69,12 @@ function assertIdFieldGraphQLTypesCorrect(list: ListForValidation) {
       `The idField on a list must define a uniqueWhere GraphQL input with the ID GraphQL scalar type but the idField for ${list.listKey} does not define one`
     );
   }
-  if (idField.input.uniqueWhere.arg.type !== schema.ID) {
```

Comment:
This check disallowing changing the type of ID field, so we can't change, for example, `autoIncrement` to `integer`. I see no reasons to keep it, so remove is right decision?